.atomist/editors/ConvertManifestToPackageJson.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2017 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Project } from "@atomist/rug/model/Project";
+import { Editor, Parameter, Tags } from "@atomist/rug/operations/Decorators";
+import { EditProject } from "@atomist/rug/operations/ProjectEditor";
+import { Pattern } from "@atomist/rug/operations/RugOperation";
+
+import * as yaml from "js-yaml";
+
+const packageJsonName = ".atomist/package.json";
+const manifestYamlName = ".atomist/manifest.yml";
+
+@Editor("ConvertManifestToPackageJson",
+    "converts an existing Rug archive manifest.yml to a package.json")
+@Tags("rug", "atomist")
+export class ConvertManifestToPackageJson implements EditProject {
+
+    public edit(project: Project) {
+        if (project.fileExists(manifestYamlName)) {
+            const manifestContent = project.findFile(manifestYamlName).content;
+            const manifestYaml = yaml.load(manifestContent);
+            const packageJson = this.loadPackageJson(project);
+
+            const newPackageJson: any = {};
+            // First we add name and version to the top of the package.json
+            newPackageJson.name = `@${manifestYaml.group}/${manifestYaml.artifact}`;
+            newPackageJson.version = manifestYaml.version;
+
+            // Then we copy all existing keys into the new package.json
+            // tslint:disable-next-line:forin
+            for (const key in packageJson) {
+                newPackageJson[key] = packageJson[key];
+            }
+
+            // Create or retrieve atomist section from package.json
+            let atomist: any = {};
+            if (newPackageJson.atomist == null) {
+                newPackageJson.atomist = atomist;
+            } else {
+                atomist = newPackageJson.atomist;
+            }
+
+            atomist.requires = manifestYaml.requires;
+
+            // Add in dependencies
+            if (manifestYaml.dependencies != null) {
+                atomist.dependencies = {};
+                for (const d of manifestYaml.dependencies) {
+                    const parts = (d as string).split(":");
+                    atomist.dependencies[`${parts[0]}:${parts[1]}`] = parts[2];
```

Comment:
Isn't version optional for a dependency in the manifest? Can our package.json parser deal with an empty value for a dependency, i.e., if `parts[2]` is null?